streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryKeyValueStore.java

```diff
@@ -167,11 +167,6 @@ public boolean hasNext() {
             return new KeyValue<>(entry.getKey(), entry.getValue());
         }
 
-        @Override
-        public void remove() {
-            iter.remove();
```

Comment:
Ditto here and elsewhere, I think it's safer to not remove the delegate call, and just remove the impl of inner most classes if it throws?

Reply:
If we want to NOT support removal from iterator at all then that makes sense. Maybe we should update the javadoc of `KeyValueIterator` saying that remove is never supported in that case?

Reply:
Yeah, none of these Iterators should be supporting remove so it doesn't make sense to delegate just to throw. But ack on updating javadocs to clarify this